contracts/EscrowMigrator.sol

```diff
@@ -0,0 +1,453 @@
+        totalRegistered += registeredEscrow;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                                 STEP 2
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The user must vest any registered entries and approve this contract to spend `toPay` amount of liquid $KWENTA
+    /// before proceeding to step 3
+    /// @notice The user MUST NOT vest any non-registered entries at this point (or they will have to pay extra)
+
+    /*//////////////////////////////////////////////////////////////
+                                 STEP 3
+    //////////////////////////////////////////////////////////////*/
+
+    /// @inheritdoc IEscrowMigrator
+    function migrateEntries(address _to, uint256[] calldata _entryIDs) external {
+        _migrateEntries(msg.sender, _to, _entryIDs);
+    }
+
+    function _migrateEntries(address _account, address _to, uint256[] calldata _entryIDs)
+        internal
+        whenNotPaused
+    {
+        if (!initiated[_account]) revert MustBeInitiated();
+        _payForMigration(_account);
+
+        uint256 migratedEscrow;
+        uint256 cooldown = stakingRewardsV2.cooldownPeriod();
+
+        for (uint256 i = 0; i < _entryIDs.length; i++) {
+            uint256 entryID = _entryIDs[i];
+
+            (uint64 endTime, uint256 escrowAmount, uint256 duration) =
+                rewardEscrowV1.getVestingEntry(_account, entryID);
+            VestingEntry storage registeredEntry = registeredVestingSchedules[_account][entryID];
+            uint256 originalEscrowAmount = registeredEntry.escrowAmount;
+
+            // if it is not zero, it hasn't been vested
+            if (escrowAmount != 0) continue;
+            // entry must have been registered
+            if (originalEscrowAmount == 0) continue;
+            // skip if already migrated
+            if (registeredEntry.migrated) continue;
+
+            /// @dev it essential for security that the duration is not less than the cooldown period,
+            /// otherwise the user could do a governance attack by bypassing the unstaking cooldown lock
+            /// by migrating their escrow then staking, voting, and vesting immediately
+            if (duration < cooldown) {
+                uint256 timeCreated = endTime - duration;
+                duration = cooldown;
+                endTime = uint64(timeCreated + cooldown);
+            }
+
+            IRewardEscrowV2.VestingEntry memory entry = IRewardEscrowV2.VestingEntry({
+                escrowAmount: originalEscrowAmount,
+                duration: duration,
+                endTime: endTime,
+                earlyVestingFee: 90
+            });
+
+            kwenta.transfer(address(rewardEscrowV2), originalEscrowAmount);
+            rewardEscrowV2.importEscrowEntry(_to, entry);
+
+            registeredEntry.migrated = true;
+
+            migratedEscrow += originalEscrowAmount;
```

Comment:
nit: I know we have control of the control flow in both these external calls, but my inner CEI itch won't go away